File: java/src/main/java/problems101_200/SingleNumber.java
package problems101_200;

import java.util.ArrayList;
import java.util.List;

enum SingleNumber {
    DEFAULT {
        @Override
        int singleNumber(int[] nums) {
            List<Integer> result = new ArrayList<>();
            for (int num : nums) {
                if (!result.contains(num)) {
                    result.add(num);
                } else {
                    result.remove(Integer.valueOf(num));
                }
            }
            return result.get(0);
        }
    };

    abstract int singleNumber(int[] nums);
}

File: kotlin/src/main/kotlin/problems101_200/SingleNumber.kt
package problems101_200

enum class SingleNumber {
    DEFAULT {
        override fun singleNumber(nums: IntArray): Int {
            val result = arrayListOf<Int>()
            for (num in nums) {
                if (num !in result) {
                    result += num
                } else {
                    result -= num
                }
            }
            return result[0]
        }
    },
    ;

    abstract fun singleNumber(nums: IntArray): Int
}
